apiserver/facades/agent/instancemutater/lxdprofilewatcher.go

```diff
@@ -550,11 +555,11 @@ func (w *machineLXDProfileWatcher) provisionedChange() error {
 		return nil
 	}
 
-	m, err := w.backend.Machine(w.machine.Id())
+	machineUUID, err := w.machineService.GetMachineUUID(context.TODO(), coremachine.Name(w.machine.Id()))
```

Comment:
It'll take a couple of steps, but this can be wired up to use a context fairly easily.

It'll have to be an attribute of the watcher.

We'll have to do it anyway as we increasingly transition this watcher over to domains

Reply:
indeed, but this is a bit out of scope for this PR which is already quite big. It can be done later.